File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque

N = int(sys.stdin.readline())
queue = deque()

for _ in range(N):
    command = sys.stdin.readline().split()

    if command[0] == "push":
        queue.append(int(command[1]))

    elif command[0] == "pop":
        if not queue:
            print(-1)
        else:
            print(queue.popleft())

    elif command[0] == "size":
        print(len(queue))

    elif command[0] == "empty":
        if not queue:
            print(1)
        else:
            print(0)

    elif command[0] == "front":
        if not queue:
            print(-1)
        else:
            print(queue[0])

    else:
        if not queue:
            print(-1)
        else:
            print(queue[-1])

File: 백준/Silver/10845. 큐/큐.java
import java.util.*;
import java.io.*;

// The main method must be in a class named "Main".
class Main {
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N = sc.nextInt();
        Deque<Integer> queue = new ArrayDeque<>();
            
        while (N-- > 0) {
            String cmd = sc.next();
            if (cmd.equals("push")) {
                int num = Integer.parseInt(sc.next());
                queue.offer(num);
            }
            else if (cmd.equals("pop")) {
                if (queue.isEmpty()) {
                    bw.write("-1\n");
                } else {
                    bw.write(queue.poll() + "\n");
                }
            }
            else if (cmd.equals("size")) {
                bw.write(queue.size() + "\n");
            }
            else if (cmd.equals("empty")) {
                if (queue.isEmpty()) {
                    bw.write("1\n");
                } else {
                    bw.write("0\n");
                }
            }
            else if (cmd.equals("front")) {
                if (queue.isEmpty()) {
                    bw.write("-1\n");
                } else {
                    bw.write(queue.peek() + "\n");
                }
            }
            else {
                if (queue.isEmpty()) {
                    bw.write("-1\n");
                } else {
                    bw.write(queue.peekLast() + "\n");
                }
            }
        }
        bw.flush();
        sc.close();
    }
}